main.go

```diff
@@ -132,10 +139,30 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(
 			up, prometheus.GaugeValue, 0,
 		)
-		log.Errorf("Failed to collect health from Vault server: %v", err)
+		e.logger.Errorf(context.Background(), err, "Failed to collect health from Vault server")
 		return
 	}
 
+	k8sAuth, err := auth.NewKubernetesAuth("healthcheck")
+	if err != nil {
+		ch <- prometheus.MustNewConstMetric(
+			up, prometheus.GaugeValue, 0,
+		)
+		e.logger.Errorf(context.Background(), err, "Failed to initialize k8s auth")
+		return
+	}
+
+	_, err = e.client.Auth().Login(context.Background(), k8sAuth)
+	if err != nil {
+		if err != nil {
```

Comment:
```suggestion
	if err != nil {
		if err != nil {
```
Looks like it's a dup 🙂

Reply:
what's wrong with me